Component configuration guides (#41)
kinvolk/lokomotive: docs/configuration-guides/components/rook-ceph.md

  # Optional arguments
  namespace = "rook-test"
  monitor_count = 3
  metadata_device = "md127"
  node_selector {
    key      = "node-role.kubernetes.io/storage"
    operator = "Exists"
  }
  node_selector {
    key      = "storage.lokomotive.io"
    operator = "In"

    # If the `operator` is set to `"In"`, `values` should be specified.
    values = [
      "foo",
    ]
  }
  toleration {
    key      = "storage.lokomotive.io"
    operator = "Equal"
    value    = "rook-ceph"
    effect   = "NoSchedule"
  }
}
```

The Ceph cluster needs to be deployed in the same namespace as the Rook operator at the moment.
Additional `Roles` and `RoleBindings` need to be created if deploying across separate namespaces is
desired.

## Argument Reference

Table of all the arguments accepted by the component.

Example:

| Argument            | Description                                                                                                                                   | Default | Required |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|:-------:|:--------:|

> | `namespace`         | Namespace to deploy the ceph cluster into. Must be same as the rook operator.                                                                 | rook    | false    |

Reviewer: ```suggestion
| `namespace`         | Namespace to deploy the ceph cluster into. Must be the same as the rook operator.                                                                 | rook    | false    |
```

> | `monitor_count`     | Number of ceph monitors to deploy. Odd number like 3 or 5 is recommended which should also be sufficient for most cases.                      | 1       | false    |

Reviewer: ```suggestion
| `monitor_count`     | Number of ceph monitors to deploy. An odd number like 3 or 5 is recommended which should also be sufficient for most cases.                      | 1       | false    |
```

| `node_selector`     | Node selectors for deploying the ceph cluster pods.                                                                                           | -       | false    |

> | `toleration`        | Tolerations that the ceph cluster's pods will tolerate.                                                                                       | -       | false    |

Reviewer: ```suggestion
| `toleration`        | Tolerations that the ceph cluster pods will tolerate.                                                                                       | -       | false    |
```

> | `metadata_device`   | Name of the device to store the metadata on each storage machine. **Note**: Provide just the name of the device and skip prefixing `/dev/`.   | -       | false    |

Reviewer: ```suggestion
| `metadata_device`   | Name of the device to store the metadata on each storage machine. **Note**: Provide just the name of the device and skip prefixing with `/dev/`.   | -       | false    |
```

## Installation

To install the Rook-Ceph component:

```bash
lokoctl component install rook-ceph
```

Once the Ceph cluster is ready, an object store can be deployed to start writing to Ceph.
More information is available at [configuring Ceph object store CRD](https://rook.io/docs/rook/v1.2/ceph-object-store-crd.html)

## Uninstallation

> After removing the component from the cluster, make sure to delete `/var/lib/rook` from hostpath of

Reviewer: ```suggestion
After removing the component from the cluster, make sure to delete `/var/lib/rook` from host filesystem of
```

| `node_selector`     | Node selectors for deploying the ceph cluster pods.                                                                                                | -       | false    |
| `toleration`        | Tolerations that the ceph cluster pods will tolerate.                                                                                              | -       | false    |
| `metadata_device`   | Name of the device to store the metadata on each storage machine. **Note**: Provide just the name of the device and skip prefixing with `/dev/`.   | -       | false    |

## Installation

To install the Rook-Ceph component:

```bash
lokoctl component install rook-ceph
```

Once the Ceph cluster is ready, an object store can be deployed to start writing to Ceph.
More information is available at [configuring Ceph object store CRD](https://rook.io/docs/rook/v1.2/ceph-object-store-crd.html)

## Uninstallation

To uninstall the component:

```bash
lokoctl component render-manifest rook-ceph | kubectl delete -f -
```

> After removing the component from the cluster, make sure to delete `/var/lib/rook` from host

Reviewer: ```suggestion
After removing the component from the cluster, make sure to delete `/var/lib/rook` from the host
```
